File: fix_app.js
const fs = require('fs');

let content = fs.readFileSync('src/App.tsx', 'utf8');

// The file ends at `export default function App() { ... }` that is duplicated.
// Let's find the FIRST export default function App() and cut everything after it.
const firstExportIndex = content.indexOf('export default function App() {');
if (firstExportIndex !== -1) {
  let cutContent = content.substring(0, firstExportIndex);
  cutContent += `export default function App() {
  return (
    <ThemeProvider>
      <CustomerAuthProvider>
        <TimeveraStore />
      </CustomerAuthProvider>
    </ThemeProvider>
  );
}
`;
  fs.writeFileSync('src/App.tsx', cutContent);
}

File: fix_app.py
import re

with open('src/App.tsx', 'r') as f:
    content = f.read()

idx = content.find('export default function App() {')
if idx != -1:
    content = content[:idx] + """export default function App() {
  return (
    <ThemeProvider>
      <CustomerAuthProvider>
        <TimeveraStore />
      </CustomerAuthProvider>
    </ThemeProvider>
  );
}
"""
    with open('src/App.tsx', 'w') as f:
        f.write(content)
